orderbook/src/database/trades.rs

```diff
@@ -295,4 +343,121 @@ mod tests {
         )
         .await;
     }
+
+    // Testing Trades with settlements
+    #[tokio::test]
+    #[ignore]
+    async fn postgres_trades_with_same_settlement() {
+        let db = Database::new("postgresql://").unwrap();
+        db.clear().await.unwrap();
+        let (owners, order_ids) = generate_owners_and_order_ids(2, 2).await;
+        assert_trades(&db, &TradeFilter::default(), &[]).await;
+
+        let settlement_hash = H256::from_low_u64_be(2);
+        db.insert_events(vec![(
+            EventIndex {
+                block_number: 0,
+                log_index: 4,
+            },
+            Event::Settlement(DbSettlement {
+                solver: H160::from_low_u64_be(2),
+                transaction_hash: settlement_hash,
+            }),
+        )])
+        .await
+        .unwrap();
+
+        let event_index_a = EventIndex {
+            block_number: 0,
+            log_index: 0,
+        };
+        let trade_a = add_order_and_trade(
+            &db,
+            owners[0],
+            order_ids[0],
+            event_index_a,
+            Option::from(settlement_hash),
```

Comment:
```suggestion
            Some(settlement_hash),
```
This happens a couple more times below.